preshed/bloom.pyx

```diff
@@ -109,8 +198,8 @@ cdef void bloom_add(BloomStruct* bloom, key_t item) nogil:
     cdef key_t one = 1 # We want this explicitly typed, because bits
     hash128_x86(&item, sizeof(key_t), 0, &keys)
     for hiter in range(bloom.hcount):
-        hv = (keys[0] + (hiter * keys[1])) % bloom.length
-        bloom.bitfield[hv // sizeof(key_t)] |= one << (hv % sizeof(key_t))
+        hv = (keys[0] + (hiter * keys[1])) % bloom.length # length is in BITS
+        bloom.bitfield[hv // KEY_BITS] |= one << (hv % KEY_BITS)
```

Comment:
Nice catch! So the old implementation only used 8 of the 64-bit in each bucket? What I don't fully understand is how this works with the legacy deserialization code above? In the old implementation hashes were dispersed over 8 bits and in the correct implementation over 64 bits, so don't we get completely different bit values? E.g. what may be a collision in 8 bits, may not be a collision in 64 bits.

Reply:
In the old code, only 8 bits of each 64 bit buckets were used, but eight times as many buckets as necessary were also used. That's why this mistake didn't affect the correctness of the bloom filter. So in the correct form, the same number of bytes are actually written to, but gaps are eliminated.

If something should have used 10 bytes, for example, it would end up using 80 bytes in memory, but only 1/8 of those bytes would be written to. If those bytes are extracted and re-assembled in contiguous order, they represent the same data as in the legacy format, just without the empty gaps. 

Kind of like this:

```
A0000000 B0000000 C0000000 D0000000
... becomes ...
ABCD
```